docs/src/test.md

```diff
@@ -46,6 +46,96 @@ pytest -svv forge/test/mlir/test_ops.py::test_add
 
 > - The `-svv` flag is optional and used to display more information about the test run.
 
+## Using `@pytest.mark.parametrize`
+
+The `@pytest.mark.parametrize` decorator allows you to run a single test function with multiple sets of inputs. This is particularly useful for testing a function with various combinations of arguments and expected results.
+
+### Syntax
+
+```python
+@pytest.mark.parametrize("arg1, arg2, expected", [
+    (1, 2, 3),
+    (2, 3, 5),
+    (3, 5, 8),
+])
+def test_addition(arg1, arg2, expected):
+    assert arg1 + arg2 == expected
+```
+
+### Explanation
+- The first argument is a comma-separated string of parameter names.
+- The second argument is a list of tuples, each containing values for the parameters.
+- Pytest runs the test function once for each tuple, substituting the parameter values.
+
+## Marking Specific Parameters with `pytest.param`
+
+You can use `pytest.param` to mark specific parameter combinations with additional metadata, such as expected failures (`xfail`).
+
+#### Syntax
+
+```python
+@pytest.mark.parametrize("inputs", [
+    pytest.param(
+        ((1, 2, 3), (4, 5, 6)), marks=pytest.mark.xfail(reason="reason"))
+])
+```
+
+#### Explanation
+- `pytest.param` allows you to add specific markers or metadata to individual parameter sets.
+- In this example, the first parameter combination is marked as `xfail` with a reason provided, indicating it is expected to fail.
+- This is useful when only some parameter sets are failing or not working correctly.
+
+## Skipping Tests with `@pytest.mark.skip`
```

Comment:
As mentioned, this one is good! However, a detailed explanation is probably redundant. We can show an example, and refer to other details in the main pytset documentation.